### PR TITLE
lpeg: new recipe

### PR DESCRIPTION
**Pull request by hanya**

### CODE REVIEW

**Comment by hanya** [comment]
Required by corsix_th 0.62, #3657.

File: dev-lua/lpeg/lpeg-1.0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,58 @@
+SUMMARY="Pattern-matching library for Lua"
+DESCRIPTION="
```
Comment (Begasus) [inline dev-lua/lpeg/lpeg-1.0.1.recipe]: Please start the description on this line :)

File: dev-lua/lpeg/lpeg-1.0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:liblua$secondaryArchSuffix >= 5.3
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:liblua$secondaryArchSuffix >= 5.3
+	lua${secondaryArchSuffix}_devel >= 5.3
```
Comment (Begasus) [inline dev-lua/lpeg/lpeg-1.0.1.recipe]: These 2 lines declare the same thing, lua${secondaryArchSuffix}_devel >= 5.3 should be obsolete (not checked)

File: dev-lua/lpeg/lpeg-1.0.1.recipe [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:lua5.3$secondaryArchSuffix
```
Comment (Begasus) [inline dev-lua/lpeg/lpeg-1.0.1.recipe]: Do you need the cmd?

Comment (hanya) [reply dev-lua/lpeg/lpeg-1.0.1.recipe]: To obtain lua version in BUILD aind INSTALL.

Comment (Begasus) [reply dev-lua/lpeg/lpeg-1.0.1.recipe]: Checked without the cmd, works ok, ps, the cmd is also pulled by lib:liblua$secondaryArchitecture (part of that package)

**Comment by Begasus** [comment]
TEST() fails atm:
```
Testing ...
./test.lua
General tests for LPeg library
version 1.0.1
+
+
+
+
+
lua: ./test.lua:1431: assertion failed!
stack traceback:
         [C]: in function 'assert'
        ./test.lua:1431: in main chunk
        [C]: in ?
makefile:44: recipe for target 'test' failed
make: *** [test] Error 1
```

File: dev-lua/lpeg/lpeg-1.0.1.recipe (hunk 3 above)
Comment (hanya) [reply dev-lua/lpeg/lpeg-1.0.1.recipe]: Removed.

**Comment by Begasus** [comment]
Nice! LGTM, test passed on lpeg_x86 (on gcc2h) (not tested on x86_64)

**Comment by Begasus** [comment]
Tested on x86_64 builds ok :+1:

**Pull request merged**